dummyserver/handlers.py

```diff
@@ -264,6 +266,17 @@ def status(self, request):
 
         return Response(status=status)
 
+    def retry_after(self, request):
+        if hasattr(TestingApp, 'last_req') and \
+                datetime.now() - TestingApp.last_req < timedelta(seconds=1):
+            return Response(
+                    status="429 Too Many Requests",
+                    headers=[('Retry-After', '1')])
+
+        TestingApp.last_req = datetime.now()
```

Comment:
I would vastly prefer not to patch this on to the class object: it is really fairly gross. Instead, it should be possible to set it on the `Application` object created in `server.py`: we can initialize it to zero in that code and then use `self.application` to query it, rather than patch onto the class.

Are you open to doing that?